java/server/src/org/openqa/selenium/remote/server/commandhandler/UploadFile.java

```diff
@@ -60,8 +64,20 @@ public void execute(HttpRequest req, HttpResponse resp) throws IOException {
           "Expected there to be only 1 file. There were: " +
           (allFiles == null ? 0 : allFiles.length)));
     } else {
+      // IE requires multiple uploads to all be from the same directory - move files into one location
+      Path uploads_dir = Files.createDirectories(tempDir.toPath().getParent().resolve("remote_uploads"));
+      Path dest = uploads_dir.resolve(allFiles[0].getName());
+
+      if (Files.exists(dest)){
+        // IE maintains a lock on previously uploaded files so we can't delete/overwrite - move it instead
```

Comment:
This is a great comment. That seems unusual, but it's great to have this explanation.